examples/all-clusters-app/esp32/main/CMakeLists.txt

```diff
@@ -49,7 +50,17 @@ idf_component_register(PRIV_INCLUDE_DIRS
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/door-lock-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ias-zone-server"
                       #${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ias-zone-client
-                      PRIV_REQUIRES chip QRCode tft spidriver bt screen-framework)
+)
+if(${IDF_TARGET} STREQUAL "esp32")
+    idf_component_register(PRIV_INCLUDE_DIRS ${PRIV_INCLUDE_DIRS_GROUPS} 
+                           SRC_DIRS ${SRC_DIRS_GROUPS}
+                           PRIV_REQUIRES chip QRCode tft spidriver bt screen-framework)
+elseif(${IDF_TARGET} STREQUAL "esp32c3")
+    idf_component_register(PRIV_INCLUDE_DIRS ${PRIV_INCLUDE_DIRS_GROUPS}
```

Comment:
Set `${PRIV_INCLUDE_DIRS_GROUPS}`  based on the `${IDF_TARGET}` and set another variable for private component dependencies. Leave only one `idf_component_register` here.